### DISCUSSION TITLE
Troubleshooting your virtual network configuration

### DISCUSSION THREAD

**Discussion by seasource** [Seagate/cortx#777] [Q&A] [answered]
Hi all,

I already configured the virtual network editor according to this page --> https://github.com/Seagate/cortx/blob/main/doc/troubleshoot_virtual_network.rst

however, it still doesn't acquire IP address. Any suggestion to fix this? 

![image](https://user-images.githubusercontent.com/77312278/108047208-7ff7c080-7080-11eb-844e-80382478f896.png)

![image](https://user-images.githubusercontent.com/77312278/108047269-8f770980-7080-11eb-869d-6058b8b3d300.png)

Also issued ifdown and ifup command on ens192, it didn't help.

![image](https://user-images.githubusercontent.com/77312278/108047653-04e2da00-7081-11eb-9a87-8ea5b54d5756.png)

**Comment by hessio** [comment]
Hi @seasource thanks for your comment, what Hypervisor are you using? If you could run `nmtui`, this will open up a pop up window, where you can select "Edit a connection" Your network connections will show up here and then you can select one of them and navigate to "Automatically connect" and press the space bar. Do this for all connects. Then select "Back" then "OK" and then type "sudo reboot now" in your terminal and when it restarts you should be able to see your ip addresses with `ip a l`

**Reply by justinzw** [reply]
@seasource another issue could be the DHCP on your router not assigning addresses on your VM. What concerns me is that there is not a single IP address on your machine.

I think the first troubleshooting step is to see whether you can get an IP address on a Vanilla CentOS image or even an Ubuntu image. 

Try downloading an image here : https://www.linuxvmimages.com/images/centos-8/ and see whether you can get an IP address.

Also like @hessio asked. What Hypervisor are you using?

**Reply by seasource** [reply]
@hessio and @justinzw 
Thanks for your response and advice.

The Hypervisor I'm using is VMWare Workstation Player 16.
I configured it to auto connect and reboot, but still do not have IP assigned to ens192.

The other system installed with Centos Linux is able to fully connected to internet, but I'm not sure how to install the image from here --> https://github.com/Seagate/cortx/releases/download/VA/cortx-va-1.0.2.zip

on Centos OS 7 setup based on this installation image: --> https://vault.centos.org/7.8.2003/isos/x86_64/CentOS-7-x86_64-DVD-2003.iso

![image](https://user-images.githubusercontent.com/77312278/108171610-ba239980-7136-11eb-889a-e0e58162cfb7.png)
![image](https://user-images.githubusercontent.com/77312278/108171641-c60f5b80-7136-11eb-8973-07866df28bf2.png)

**Reply by hessio** [reply]
If I understand correctly you are trying to install this iso file onto a booted VM? You cannot do this, what you need to do is unzip this file. Once the file is unzipped use VMWare Workstation Player to open this iso file. 

> The other system installed with Centos Linux is able to fully connected to internet, but I'm not sure how to install the image from here --> https://github.com/Seagate/cortx/releases/download/VA/cortx-va-1.0.2.zip

**Reply by hessio** [reply]
Also if you would like to we can have a call and I can walk you through what I understand because sometimes the messages can be difficult to convey your issues and I will post the problem/solution here then for anyone else who may run into a similar issue for them here? Email: patrick.hession@seagate.com

**Reply by seasource** [reply]
@hessio 
We isolated this issue by using the same setup at home, found it is because corporate network preventing the connection. 
Since I can't use VM for using office desktop, I will need instructions to deploy Cortx on an independent Linux PC.
I tried follow the suggestions in this post --> https://github.com/Seagate/cortx/discussions/575
however, encountered multiple script errors and compile/build errors.

I couldn't find a complete set of instruction or guide for deploying Cortx on a Linux/Centos PC.

**Reply by hessio** [reply]
Oh sorry to hear this @seasource, I am glad you found a workaround for this network issue. What are the script errors and compile/build errors you are seeing? I will try to recreate these issues in my own setup and try my best to unblock you, if I can. 

Unfortunately I think that you need specific kernel versions of CentOS, can you run `uname -r` to see what kernel version you are using, also which version of CentOS are you using `cat /etc/redhat-release` ?

**Reply by seasource** [reply]
hi @hessio ,
Thanks for your reply. 
I'm now looking at this entry --> https://github.com/Seagate/cortx/blob/main/QUICK_START.md
It appears that I do not have adequate hardware to do this "Distributed Scaled-out CORTX" --> https://github.com/Seagate/cortx/blob/main/doc/scaleout/README.rst
It requires at least 3 network port per system and 3 systems are required, in addition to several other 'heavy weight' hardware specs.
Unfortunately Single Node CORTX setup guide is currently not documented as per this entry --> https://github.com/Seagate/cortx/blob/main/QUICK_START.md

In view of these challenges, I will have but to postpone any further activity on CORTX at this time.

Thanks for your help.

**Reply by hessio** [reply]
Hi @seasource I am really sorry to hear this, I wish I could have been more help to you! 

If it is any consolation you can build Motr and HARE on bare metal following this guide to build [Motr](https://github.com/Seagate/cortx-motr/blob/main/doc/Quick-Start-Guide.rst) and this guide to build [HARE](https://github.com/Seagate/cortx-hare/blob/main/README.md). This was a great learning experience for me and I would be more than happy to work through it with you.

HARE is used to stop/start and configure Motr nodes so we will need this to run your motr instance and motr is the engine of the entire system. 

Currently we don't have any guide for a way to run the entire CORTX system on bare metal, but we are close!

I would be more than happy to help out with any issues you have while doing any of these processes. If not, thanks for trying to get involved and sorry once again that we weren't able to help you out.

Also how much RAM do you have, do you have more than 8GB?

And as always, my email patrick.hession@seagate.com is here if you want to organize to have a call.

**Reply by seasource** [reply]
Hi @hessio 
Thanks for your kind words and the very helpful information from you.
The system I'm using now has 8GB RAM installed.
Following are my system Centos kernel and version.
- CentOS Linux release 7.9.2009 (Core)
- 3.10.0-1160.15.2.el7.x86_64

**Reply by hessio** [reply]
Just glad I could help you out. Feel free to ask as many questions as possible because others may be able to learn from them too.

Yes this is good you should be able to build and run Motr + HARE with this system.

**Comment by mukul-seagate11** [comment]
@seasource, 

firstly, you have to set the network configuration to NAT
Secondly, you can check the network configuration file in OS & modify the "ONBOOT=yes" & then reboot the VM,
 /etc/sysconfig/network-scripts/ifcfg-ens192
/etc/sysconfig/network-scripts/ifcfg-ens224
/etc/sysconfig/network-scripts/ifcfg-ens256

**Reply by andriytk** [reply]
@mukul-seagate11, any idea why in the mentioned above troubleshooting doc it states:

> Edit your virtual machine settings and set every network adapter to use a Bridged connection and to 'Replicate physical network connection state':

?

**Comment by mukul-seagate11** [comment]
@VenkyOS, please check & correct as the OVA configuration should be NAT and not bridge?

**Reply by justinzw** [reply]
@mukul-seagate11 I don't think it matters. I used bridged and mine works fine.

**Reply by mukul-seagate11** [reply]
> @mukul-seagate11 I don't think it matters. I used bridged and mine works fine.

OK than Its good

**Comment by johnbent** [comment, marked as answer]
Hello @seasource .  Thanks for checking us out and sorry if it has been slower than it should.  However, according to @shailesh-vaidya , this document https://github.com/Seagate/cortx-prvsnr/wiki/Auto-Deploy-VM-Provisioner-CLI-Commands should be what you're looking for if you're up for trying one more thing?  :)

**Reply by seasource** [reply]
Hi @johnbent 
I'm currently having problems with VM based setup due to constraint related to corporate network.
Is it possible to deploy a full version CORTX on Centos OS on a low cost PC with limited CPU power and RAM for the purpose of basic evaluation, testing and some development work?
Thanks in advance for your help

**Reply by hessio** [reply]
Unfortunately we wouldn't have support for this right now, you can run Motr and HARE on their own (steps are mentioned in the above thread reply). There are some bugs with this provisioner link John has sent. But the team are working on this and are very close to having this finished. You would be able to run performance tests, development, basic evaluation on Motr + HARE only.

I will update this discussion when the Provisioner team have completed their update for this iso so you will get a notification when it is possible to build entire CORTX on single node on bare metal. For now you can build Motr + HARE on bare metal (as outlined in above thread).

Thanks so much for these questions and we are always happy to try help you out. Hope this helps you!

**Reply by seasource** [reply]
Hi @hessio 

Appreciate your guidance and patience for me.
Based on the steps in the above thread from your reply, I have succeeded in compiling Motr, executing unit tests and also running system single node test. All unit tests run look ok, however noted some errors for system single node test - "sudo ./scripts/m0 run-st 52motr-singlenode-sanity"

![image](https://user-images.githubusercontent.com/77312278/108693399-c9ef0500-74cb-11eb-997a-0622a0f33a1a.png)

Is this expected?
 
Next I will compile Hare and execute some test.
If all goes well, I'll try to see where I may contribute to CORTX from there..

Thanks again for your help.

**Reply by hessio** [reply]
Hi @seasource, there is no harm in this error you can keep proceeding along with the steps. In the HARE guide when you are asked to build and install Motr you will not need to repeat the steps you have done in this Motr guide, only the steps you haven't done you will need to do. 

Great to hear, really happy to see you are making progress on the setup guides too and you are looking to contribute too! Here is a link to some [CORTX stickers](https://www.seagate.com/gb/en/promos/cortx-stickers/) if you would like us to send some out to you!
